### DISCUSSION TITLE
The maximum value engine upgrade is ready - when we should deploy?

### DISCUSSION THREAD

**Discussion by ipaleka** [asastats/channel#95] [General]
Give your suggestions about the time and eventual announcements.

Yieldly announced ASASTATS staking pools for tomorrow/Monday and we should probably time that in regard to this upgrade: what if Yieldy's staking pools are going to be deployed tomorrow and what if they aren't?

**Comment by wxivd** [comment]
Congratulations for all of us. This is another milestone and I am really excited for this. Thank you

I think Monday is ideal. 6PM London time = 1PM New York time is worth considering. 

<img width="1070" alt="Screenshot 2022-05-22 at 22 01 28" src="https://user-images.githubusercontent.com/95645978/169715643-f8d2c06c-b41c-4ad3-9939-7decb58e5553.png">


> Yieldy's staking pools are going to be deployed tomorrow and what if they aren't?

These two are separate announcements. If they happen on the same day that is a plus. If not, there is still enough reason to celebrate. 

In terms on magnitude, I see engine upgrade is more important. 

Yieldly announcement is welcome whenever it happens. I looked at their previous announcements and there is no regular time slot as per their Twitter account.

**Reply by ipaleka** [reply]
> Congratulations for all of us. This is another milestone and I am really excited for this. Thank you

Yes, this is basically a cornerstone for our API. The thing is that this involves an order of magnitude more calculations so we had to optimize before implementation. And the outcome is that the increase in speed averages above 100%. :)
 
> I think Monday is ideal. 6PM London time = 1PM New York time is worth considering.
 
> > Yieldy's staking pools are going to be deployed tomorrow and what if they aren't?
> 
> These two are separate announcements. If they happen on the same day that is a plus. If not, there is still enough reason to celebrate.
> 
> In terms on magnitude, I see engine upgrade is more important.

Ok, so your stance is that we should go tomorrow 6PM London time regardless of Yieldly. Thanks.
 
> Yieldly announcement is welcome whenever it happens. I looked at their previous announcements and there is no regular time slot as per their Twitter account.

Before the first delay a few months ago they used to deploy at 01:30 CET, while in the last period that seems to be 07:00 CET (15:00 in Australia).

**Comment by ipaleka** [comment]
by Mayor | CM YouTube in [Discord](https://discord.com/channels/906917846754418770/965692939521097758/978048383698731058):

- - - - -

Probably not the correct channel but - I agree with 1pm EST deployment.  That’s the most active “drop” time for NFTs and there is a lot of face-to-screen activity around that time

- - - - -

**Comment by ipaleka** [comment]
We used to have these announcements on Reddit, and then we linked them to Twitter and Discord. I suppose we should switch here from now on - we already have got that "Announcements" category. :)

**Reply by wxivd** [reply]
This is our official channel so project announcements are expected to be here. 

A summary or short statement is suitable for different announcement channels (Twitter, Reddit, Discord) along with a link to the announcement here.

**Reply by ipaleka** [reply]
> This is our official channel so project announcements are expected to be here.
> 
> A summary or short statement is suitable for different announcement channels (Twitter, Reddit, Discord) along with a link to the announcement here.

Twitter and Discord use link text beginning snippet after a link is posted, so that goal can probably be satisfied by creating the first paragraph with that in mind. 

A link post on Reddit can probably be a better option than a text post imo. Redditors will either click a post or they will not. xD So for Reddit we can eventually extend the original Channel's title into a very short summary, Btw, Reddit uses the first image found under the link to present it as a post's thumbnail.

**Reply by wxivd** [reply]
@ipaleka I suggest the following announcement. Please edit as it's deemed necessary. 

### The maximum value engine upgrade is live.

**ASA Stats has reached another milestone in its API Development. Today we instigate our newest ASA price optimisation algorithms.**

ASA Stats brings maximum value for user's asset by calculating the maximum value of the ASA held in their wallet as well as the optimal amounts that should be sold on each of the AMMs or DEXes to the users.

**What is the optimum price and Why is price optimisation important?**

- The optimal price is that price point at which the total profit of the seller is maximised. 
- When the price is too low, the seller is moving a large number of units but is not earning the highest possible aggregate profit.

**What is price optimisation?**

- Price optimisation is the process of finding the optimal value of given ASA. 
- The current price of ASA is retrieved from all the available AMMs and DEXes in a way that brings the maximum value for the user asset. 
- This is achieved by calculating current liquidities and/or volumes for that asset on all AMMs and DEXes, and by providing the optimal amounts that should be sold on each of the AMMs or DEXes to the users.

----

Maybe someone who has access to the staging environment can take some explainer photos. It would be ideal if we can pick an asset that has different LP volumes across 2-3 AMMs.

**Reply by ipaleka** [reply]
Mate, tbh, this is more than I have expected. :)

Those screenshots will be ready, but the thing is that I'm now in the middle of some changes as we have delayed deployment from yesterday, so I am not sure would the deployment be ready as planned (40 minutes from now).

**Reply by wxivd** [reply]
There is no need to rush announcement. Please take your time. Once ready we go announce

**Reply by ipaleka** [reply]
> There is no need to rush announcement. Please take your time. Once ready we go announce

We're ready for the deployment now :), but I'm still thinking should we wait until tomorrow?

Initial thoughts were based on Yieldly's program deployment early in the morning (CET/UTC), but that happened almost 12 hours later in the day.

**Reply by wxivd** [reply]
> but I'm still thinking should we wait until tomorrow? 

Yes, tomorrow sounds ideal. We are building momentum already, let it ride.

**Reply by ipaleka** [reply]
> Yes, tomorrow sounds ideal. We are building momentum already, let it ride.

Great! So the planned time (6PM London time), but a day after.

**Reply by Babbexx22** [reply]
Tomorrow, it is!

**Reply by wxivd** [reply]
Explainer images are pending @ipaleka. Once these are provided.

### Announcements process:

1. Github announcement
- Post announcement in [Discussions](https://github.com/asastats/channel/discussions) according to the text in this [comment](https://github.com/asastats/channel/discussions/95#discussioncomment-2805675). 
- Set category to “**Announcements**”. I can’t find this category available, I presume it has something to do with access permission?

2. Twitter announcement

- ASA Stats has reached another milestone in its API Development. Today we instigate our newest ASA price optimisation algorithms. 
#Algorand #algofam #asastats
- Copy announcement link from Github (1)
- [Twitter card validator](https://cards-dev.twitter.com/validator) is a useful too to preview haw the tweet would look like before posting it. 
- Preview tweet before posting

3. Reddit announcement
- Similar to 1
- add link to ASA stats Website https://www.asastats.com/

**Reply by ipaleka** [reply]
Sry, this should have been posted earlier in the day. We will use your comment for the future announcements tho.

**Reply by ipaleka** [reply]
> * Set category to “**Announcements**”. I can’t find this category available, I presume it has something to do with access permission?

Do you have a screen similar to this when you're trying to start a new discussion?

![category](https://user-images.githubusercontent.com/49662536/170094176-e6022ddc-b2cf-4ed4-bb05-81bb8b477c8f.png)

> * [Twitter card validator](https://cards-dev.twitter.com/validator) is a useful too to preview haw the tweet would look like before posting 

Bookmarked!! Thanks.

**Reply by wxivd** [reply]
> Do you have a screen similar to this when you're trying to start a new discussion?

This is how it appears to me

<img width="1670" alt="Screenshot 2022-05-24 at 18 22 12" src="https://user-images.githubusercontent.com/95645978/170095490-80a5eb18-5faa-4159-b8fd-c430d531b2ad.png">

**Reply by ipaleka** [reply]
Ok, so the default permission (being "Triage") doesn't allow you to set that category.

I just added you only for the testing purpose under "Maintain" permission. You should check if such permission is suitable for all the DAO members or not. If it is then I'll just switch the DAO team to that permission level. If not, then I'll be the person for now who creates the announcements.

**Reply by wxivd** [reply]
I reviewed the [Permissions for each role](https://docs.github.com/en/organizations/managing-access-to-your-organizations-repositories/repository-roles-for-an-organization)

There are some technical aspects, permission wise, that falls beyond my level of experience. 

It is best for you to be the person for now who creates the announcements.

**Reply by ipaleka** [reply]
> I reviewed the [Permissions for each role](https://docs.github.com/en/organizations/managing-access-to-your-organizations-repositories/repository-roles-for-an-organization)
> 
> There are some technical aspects, permission wise, that falls beyond my level of experience. 
> 
> It is best for you to be the person for now who creates the announcements. 

Ok, thanks. We have a year or so in front of us to find a solution for this and other similar issues.

There will exist an entity (organization or person/people) in the DAO phase responsible for the servers administration. I guess something similar can be implemented for this if we can't manage to have the same permission to all members of the DAO.
